### ISSUE TITLE
KD loss function

### ISSUE THREAD

**Issue by jl749** [jl749/knowledge-distillation-pytorch#4]
https://github.com/jl749/knowledge-distillation-pytorch/blob/4a7c13d090554961f04b1b5f4ecc324cdca94d66/mnist/distill_mnist.py#L96-L97

![image](https://user-images.githubusercontent.com/67103130/164506356-c93b426a-91f9-4b41-9f67-c7dee736de88.png)

## why x2 ?
![image](https://user-images.githubusercontent.com/67103130/164506669-82bda242-6991-4e8c-9d92-cac19e054745.png)
from `/knowledge-distillation-pytorch/issues/10`

**Comment by jl749** [comment]
## why `KLDivLoss` instead of `CrossEntropyLoss` ?
pytorch CrossEntropy takes "one hot vector" as a target
which means we cannot use `CrossEntropyLoss` with soft target

THEREFORE, use `KLDivLoss` instead (it can take prob distribution as target)
